spec.html

```diff
@@ -7680,6 +7680,89 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
+      <emu-note>
+        <p>For example, WHATWG HTML uses the host-defined value to propagate the incumbent settings object for Promise callbacks.</p>
+      </emu-note>
+      <p>JobCallback Records have the fields listed in <emu-xref href="#table-jobcallback-records"></emu-xref>.</p>
+      <emu-table id="table-jobcallback-records" caption="JobCallback Record Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Callback]]
+              </td>
+              <td>
+                A callable object
+              </td>
+              <td>
+                The callback to invoke when the Job is invoked.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[HostDefined]]
+              </td>
+              <td>
+                Any, default value is *undefined*.
+              </td>
+              <td>
+                Field reserved for use by hosts.
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </emu-table>
+    </emu-clause>
+
+    <emu-clause id="sec-hostmakejobcallback" aoid="HostMakeJobCallback">
+      <h1>HostMakeJobCallback ( _callable_ )</h1>
+      <p>HostMakeJobCallback is a host-defined abstract operation that takes argument _callable_ (a callable object).</p>
+      <p>The implementation of HostMakeJobCallback must conform to the following requirements:</p>
+      <ul>
+        <li>It must always complete normally (i.e., not return an abrupt completion).</li>
+        <li>It must always return a JobCallback Record whose [[Callback]] field is _callable_.</li>
+      </ul>
+      <p>The default implementation of HostMakeJobCallback performs the following steps when called:</p>
+      <emu-alg>
+        1. Assert: IsCallable(_callable_) is *true*.
+        1. Return the JobCallback Record { [[Callback]]: _callable_, [[HostDefined]]: *undefined* }.
+      </emu-alg>
+      <emu-note>
+        <p>This is called at the time that the callback is passed to the function that is responsible for its being eventually scheduled and run, i.e. at its origination. For example, `Promise.prototype.then(thenAction)` calls MakeJobCallback on `thenAction` at the time of invoking `Promise.prototype.then`, not at the time of scheduling the reaction Job.</p>
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-hostcalljobcallback" aoid="HostCallJobCallback">
+      <h1>HostCallJobCallback ( _jobCallback_, _V_ [ , _argumentsList_ ] )</h1>
```

Comment:
It seems this could be made slightly easier on hosts, including the default-implementation host and the conformance requirement text a few lines down, if you always supplied argumentsList (sometimes empty).

Reply:
Done. Made a non-host hook convenience wrapper CallJobCallback that is optional in _argumentsList_.